scripts/biomedical/humanMetabolomeDatabase/format_metabolite.py

```diff
@@ -0,0 +1,285 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+'''
+Author: Suhana Bedi
+Date: 07/10/2021
+Name: format_metabolite.py
+Description: Add dcids for all the metabolites in the VMH database.
+Extract Chembl ids from kegg and chebi matches, and query datacommons
+to check for pre-existing nodes for the same metabolite. In addition,
+extract hmdb ids from the Human Metabolome database and use it as a
+dcid when chembl is absent.
+@file_input: input .tsv from VMH with metabolite list, and .csv from hmdb
+@file_output: csv output file with addition chembl and dcid columns
+'''
+
+import sys
+import pandas as pd
+import numpy as np
+import datacommons as dc
+from bioservices import UniChem
+import re
+
+
+def find_overlapped_id(df1, df2, df1_column, df2_column, id_return_column):
+    """find overlapped id between 2 columns from 2 dataframes
+    Args:
+        df1: pd.DataFrame 1
+        df2: pd.DataFrame 2
+        df1_column: name of column in df1 that needs to be matched
+        df2_column: name of column in df2 that needs to be matched
+        id_return_column: column used to return matched id
+    Returns:
+        panda.Series with matched ids
+    """
+    matched_id = df1[~df1[df1_column].isna()].merge(\
+    df2[~df2[df2_column].isna()], left_on=df1_column,\
+                                       right_on=df2_column)[id_return_column]
+    matched_id = matched_id.drop_duplicates()
+    return matched_id
+
+
+def name_map(dfm, dfh):
+    """
+    Finds the rows with the same name in the HMDB
+    and VMH dataset.
+    Args:
+        dfm = VMH data, dfh = HMDB data
+    Returns:
+        VMH data with added HMDB ids
+    """
+    dfm['fullName'] = dfm['fullName'].map(
+        lambda x: re.sub(r'\W+', '', x)).str.lower()
+    dfh['name'] = dfh['name'].map(lambda x: re.sub(r'\W+', '', x)).str.lower()
+    dfh['name'] = dfh['name'].str[2:]
+    dfh['name'] = dfh['name'].str[:-1]
+    overlapped_name = find_overlapped_id(dfm, dfh, "fullName", "name",
+                                         "fullName")
+    name_hmdb_dict = dfh[dfh["name"].\
+    isin(overlapped_name)][["name","accession"]].set_index("name").to_dict()["accession"]
+    dfm["hmdb"] = dfm["hmdb"].fillna(dfm["fullName"].map(name_hmdb_dict))
+    return dfm
+
+
+def kegg_map(dfm, dfh):
+    """
+    Finds the rows with the same KEGG id in the HMDB
+    and VMH dataset.
+    Args:
+        dfm = VMH data, dfh = HMDB data
+    Returns:
+        VMH data with added HMDB ids
+    """
+    overlapped_kegg = find_overlapped_id(dfm, dfh, "keggId", "kegg", "keggId")
+    kegg_hmdb_dict = dfh[dfh["kegg"].\
+    isin(overlapped_kegg)][["kegg","accession"]].set_index("kegg").to_dict()["accession"]
+    dfm["hmdb"] = dfm["hmdb"].fillna(dfm["keggId"].map(kegg_hmdb_dict))
+    return dfm
+
+
+def chebi_map(dfm, dfh):
```

Comment:
chebi_map and kegg_map both seem to be doing the same thing, but just using different column names. Can just have one method where the column names get passed in as arguments. Or can have a helper method that each of chebi_map and kegg_map calls so that we don't duplicate the code.